components/script/dom/document.rs

```diff
@@ -2399,10 +2401,98 @@ impl Document {
         }
     }
 
+    /// <https://drafts.csswg.org/cssom-view/#document-run-the-scroll-steps>
+    pub(crate) fn run_the_scroll_steps(&self, can_gc: CanGc) {
+        // Step 1.
+        // > Run the steps to dispatch pending scrollsnapchanging events for doc.
+        // TODO(#7673): Implement scroll snapping
+
+        // Step 2
+        // > For each item target in doc’s pending scroll event targets, in the order they
+        // > were added to the list, run these substeps:
+        for target in self.pending_scroll_event_targets.borrow().iter() {
+            // Step 2.1
+            // > If target is a Document, fire an event named scroll that bubbles at target.
+            if target.downcast::<Document>().is_some() {
+                target.fire_bubbling_event(Atom::from("scroll"), can_gc);
+            }
+
+            // Step 2.2
+            // > Otherwise, fire an event named scroll at target.
+            if target.downcast::<Document>().is_some() {
+                target.fire_bubbling_event(Atom::from("scroll"), can_gc);
+            }
```

Comment:
This seems like a logic error, since it's the same code as above.

Reply:
Nice catch! Thanks!